mirdata/mridangam_stroke.py

```diff
@@ -0,0 +1,262 @@
+
+"""
+Mridangam Stroke Dataset Loader
+
+The Mridangam Stroke dataset is a collection of individual strokes of
+the Mridangam in various tonics. The dataset comprises of 10 different
+strokes played on Mridangams with 6 different tonic values. The audio
+examples were recorded from a professional Carnatic percussionist in a
+semi-anechoic studio conditions by Akshay Anantapadmanabhan.
+
+Total audio samples: 6977
+
+Used microphones:
+* SM-58 microphones
+* H4n ZOOM recorder.
+
+Audio specifications
+* Sampling frequency: 44.1 kHz
+* Bit-depth: 16 bit
+* Audio format: .wav
+
+The dataset can be used for training models for each Mridangam stroke. The
+presentation of the dataset took place on the IEEE International Conference
+on Acoustics, Speech and Signal Processing (ICASSP 2013) on May 2013.
+You can read the full publication here: https://repositori.upf.edu/handle/10230/25756
+
+Mridangam Dataset is annotated by storing the informat of each track in their filenames.
+The structure of the filename is:
+<TrackID>__<AuthorName>__<StrokeName>-<Tonic>-<InstanceNum>.wav
+
+The dataset is made available by CompMusic under a Creative Commons
+Attribution 3.0 Unported (CC BY 3.0) License.
+
+For more details, please visit: https://compmusic.upf.edu/mridangam-stroke-dataset
+"""
+
+import os
+import librosa
+
+from mirdata import download_utils
+from mirdata import jams_utils
+from mirdata import track
+from mirdata import utils
+
+
+DATASET_DIR = 'Mridangam-Stroke'
+
+
+REMOTES = {
+    'remote_data': download_utils.RemoteFileMetadata(
+        filename='mridangam_stroke_1.5.zip',
+        url='https://zenodo.org/record/4068196/files/mridangam_stroke_1.5.zip?download=1',
+        checksum='39af55b2476b94c7946bec24331ec01a',  # the md5 checksum
+        destination_dir=None,  # relative path for where to unzip the data, or None
+    ),
+}
+
+DATA = utils.LargeData(
+    'mridangam_stroke_index.json'
+)
+
+
+STROKE_DICT = {
+    'bheem', 'cha', 'dheem', 'dhin', 'num', 'ta', 'tha', 'tham', 'thi', 'thom'
+}
+
+
+TONIC_DICT = {
+    'B', 'C', 'C#', 'D', 'D#', 'E'
+}
+
+
+class Track(track.Track):
+    """Mridangam Stroke track class
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets/Mridangam-Stroke`
+    Attributes:
+        track_id (str): track id
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError('{} is not a valid track ID in Example'.format(track_id))
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+        self._track_paths = DATA.index[track_id]
+
+        self.audio_path = os.path.join(self._data_home, self._track_paths['audio'][0])
+
+    @utils.cached_property
```

Comment:
see the below comment, but both of these cached properties can just be defined in the __init__ and can be removed here :)